### PR TITLE
[QoL] Improve shop readibility

### PR DESCRIPTION
**Pull request by CodeTappert**
The changes are basically made by @Dakurei 
We had other planes but changed it to this what it is know

## What are the changes?

Change colors for better contrast score (makes reading easier)
Corrects colors that were totally white when they should have been f8f8f8 like the others
Created a new overlay especially for the shop
This one is an almost black color with a higher opacity. This makes the shop elements easier to read, as they blend much less into the UI underneath

## Why am I doing these changes?

The Read is REALLY hard to read. See https://discord.com/channels/1125469663833370665/1251535006669733959

## What did change?

See screenshots or title

###Files changed
src/battle-scene.ts
Add an overlay by duplicating it from a pre-existing one
Added methods for displaying and hiding it, duplicating those already present

src/ui/modifier-select-ui-handler.ts
Replacing the old showFieldOverlay and hideFieldOverlay methods with the newly created ones

src/ui/text.ts
Corrects all-white #ffffff colors to #f8f8f8 as they should have been from the start
Corrects the color of tiers so that the contrast score is sufficient to improve legibility

### Screenshots/Videos

Korean Main:
image
Korean with this PR:
image

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

## Additional
Modifications requested by @brain-frog

### Changes
 - Add a filter on modifiers displayed in the shop to hide those held by the player's Pokémon
 - Add documentation on methods created and modified for this purpose
 - Longer fade-out time for new shop overlay (from 250ms to 750ms, the same as the fade-in time)

### Files changed
- `src/battle-scene.ts`
  > Add a `getModifierBar()` method to retrieve battle-scene's ModifierBar (which is private) from other interfaces
- `src/ui/modifier-select-ui-handler.ts`
  > Use of the modified `updateModifiers()` method to hide modifiers held by the player's Pokémon during the Shop phase
  > New overlay fade-out time modified (from 250ms to 750ms)
- `src/modifier/modifier.ts`
  > Added a `hideHeldItems` boolean parameter to hide modifiers held by the player's Pokémon when the ModifierBar is updated

### CODE REVIEW

File: src/ui/modifier-select-ui-handler.ts [hunk 1]
```diff
@@ -567,17 +567,29 @@ class ModifierOption extends Phaser.GameObjects.Container {
       this.itemContainer.add(this.itemTint);
     }
 
-    this.itemText = addTextObject(this.scene, 0, 35, this.modifierTypeOption.type.name, TextStyle.PARTY, { align: "center" });
+    this.itemText = addTextObject(this.scene, 0, 35, this.modifierTypeOption.type.name, TextStyle.PARTY, {    align: "center"   });
```
Comment (CodeTappert) [inline src/ui/modifier-select-ui-handler.ts:570]:
```suggestion
    this.itemText = addTextObject(this.scene, 0, 35, this.modifierTypeOption.type.name, TextStyle.PARTY, { align: "center" });
```

**Comment by CodeTappert** [comment]
I am willing to not do the color changes. But the shadow imo is a MUST

File: src/ui/text.ts [hunk 2]
```diff
@@ -235,15 +235,15 @@ export function getModifierTierTextTint(tier: ModifierTier): integer {
   case ModifierTier.COMMON:
     return 0xffffff;
   case ModifierTier.GREAT:
-    return 0x3890f8;
+    return 0x5da3f9;
   case ModifierTier.ULTRA:
     return 0xf8d038;
   case ModifierTier.ROGUE:
-    return 0xd52929;
+    return 0xE38787;
```
Comment (Dakurei) [inline src/ui/text.ts:242]:
```suggestion
    return 0xe38787;
```

File: src/ui/text.ts [hunk 3]
```diff
@@ -235,15 +235,15 @@ export function getModifierTierTextTint(tier: ModifierTier): integer {
...
   case ModifierTier.MASTER:
-    return 0xe020c0;
+    return 0xEC74D8;
```
Comment (Dakurei) [inline src/ui/text.ts:244]:
```suggestion
    return 0xec74d8;
```

File: src/ui/text.ts [hunk 4]
```diff
@@ -235,15 +235,15 @@ export function getModifierTierTextTint(tier: ModifierTier): integer {
...
   case ModifierTier.LUXURY:
-    return 0xe64a18;
+    return 0xEE825D;
```
Comment (Dakurei) [inline src/ui/text.ts:246]:
```suggestion
    return 0xee825d;
```

**Review by Dakurei — CHANGES_REQUESTED** [review]
Just for consistency with the other constants

**Review by Dakurei — APPROVED** [review]
This should greatly improve legibility for the majority of players

And as @CodeTappert hinted, designing an option for people with color difficulties ( i.e. protanopia, deuteranopia, tritanopia, ...) should be considered, but that's a whole other much bigger job that applies not just to the shop, but to the whole game 👍

**Review by torranx — APPROVED** [review]
approving this so it gets traction

**Comment by brain-frog** [comment]
Would it be possible for you to look at putting the overlay above the modifier bars so the modifiers arent interfering with reading the top row of the shop? I know it becomes a lot during endless

**Comment by Dakurei** [comment]
> Would it be possible for you to look at putting the overlay above the modifier bars so the modifiers arent interfering with reading the top row of the shop? I know it becomes a lot during endless

When I made the modifications for this new black overlay, I was made to understand that people might want to see what their Pokémon already has on the screen, so we'll have to choose whether we want to hide them or not

**Comment by CodeTappert** [comment]
> > Would it be possible for you to look at putting the overlay above the modifier bars so the modifiers arent interfering with reading the top row of the shop? I know it becomes a lot during endless
> 
> When I made the modifications for this new black overlay, I was made to understand that people might want to see what their Pokémon already has on the screen, so we'll have to choose whether we want to hide them or not
> 
> _EDIT: Personally, I think it would be better to hide the modifiers during the shop stage. Furthermore, as the overlay is defined in the context of the `fieldUI`, it wouldn't be easy to ensure that the modifiers (which are part of the `uiContainer`) are displayed underneath it without reviewing the structure of these overlays in a little more depth._

I agree with the idea to hide the modifiers during shop

**Review by XGSleepWalker — APPROVED** [review]

**Review by XGSleepWalker — APPROVED** [review]

**Pull request merged**